### DISCUSSION TITLE
how to detect and update multiple places with current version

### DISCUSSION THREAD

**Discussion by ghost** [renovatebot/renovate#22999] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

I've implemented artifactory search api datasource and the files with dependencies are constructed as follows (simple text file for iteration and calling `apt-get install <line> --allow-downgrades` )

```apt.requirement.txt
librsc-common=2.0.0
librsc2.0=2.0.0
librsc2.0:i386=2.0.0
```

The first line is working, but the second and third should also detect and replace the optional `<MAJ.MIN>`  in `<LIB-NAME><MAJ.MIN>=<VERSION>` lib*.so notation. For searching `depName` without `2.0` is enough but if new version is found the `<MAJ.MIN>` part should be replaced if it was defined before.

```
    "regexManagers": [
        {
            "fileMatch": ["apt.requirements.txt", "apt/requirements.txt", "libs/requirements.txt"],
            "matchStrings": ["(?<depName>.*?)(:(?<depArch>.*))?=(?<currentValue>.*)\\s"],
            "extractVersionTemplate": "^.*\/{{{depName}}}_(?<version>.*?)_{{{depArch}}}.*$",
            "datasourceTemplate": "artifactory-search",
            "registryUrlTemplate": "https://my-artifactory-server",
            "packageNameTemplate": "repos=my-artifactory-repository&name={{{depName}}}*{{{depArch}}}"
        }
    ],
```

See the logs below.

### What you tried so far.

I've searched examples and did not found a solution how to use multiple places with versioning.

### Relevant debug logs

<details><summary>Logs</summary>

```
             {
               "depName": "librsc2.0",
               "packageName": "repos=my-artifactory-repository&name=librsc2.0*i386",
               "currentValue": "2.0.0",
               "datasource": "artifactory-search",
               "extractVersion": "^.*/librsc2.0_(?<version>.*?)_i386.*$",
               "registryUrls": ["https://captain.rtf.siemens.net/"],
               "replaceString": "librsc2.0:i386=2.0.0\n",
               "updates": [],
               "versioning": "regex:^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)(?:(-|~|\\+)(?<prerelease>.*))?$",
               "warnings": [],
               "registryUrl": "https://my-artifactory-server",
               "currentVersion": "2.0.0",
               "fixedVersion": "2.0.0"
             },
```

</details>

**Comment by rarkins** [comment]
I think you need at least one more regex manager for that (treating `2.0` as the `currentValue`). Either that or a post upgrade task which does some type of search/replace based on the first line

**Reply by ghost** [reply]
Thanks, i've managed to add second regex manager, but now the two managers are finding also some same dependencies and i see this messages

```
INFO: Cannot find replaceString in current file content. Was it already updated?
```

I works, but anyway try to fix that two managers are not finding same lines...

**Reply by rarkins** [reply]
Do you want all three lines to be in sync? If so then a post upgrade task may be best
